File: sample-api/npm/src/startup.ts
import express from "express";
import { Container } from "inversify";
import { IQueries, TYPES as SampleServices, IAddSomethingHandler, IUpdateSomethingHandler, UpdateSomethingCommand, AddSomethingCommand } from "sample-services";
import { SampleApiModule } from "./dependency-injection/container-module";

export class Startup {

    private readonly _configuration: Map<string, any>;
    //private readonly _environment: string;

    public constructor(configuration: Map<string, any>, environment: string) {
        this._configuration = configuration;
        //this._environment = environment;
    }

    public configureServices(services: Container) {
        const postgresConnection = this._configuration.get("postgres");
        services.load(new SampleApiModule(options => {
            options.domainCommands = true;
            options.domainQueries = true;
            options.postgres = postgresConnection;
        }));
    }

    public configure(app: express.Application, services: Container, env: string): express.Application {

        app.get('/', async (req, res) => {
            try {
                
                const queries = services.get<IQueries>(SampleServices.IQueries);
                let records = await queries.getSomethingsAsync();
                if (records.length) {
                    const firstSomething = records[0];

                    const updateSomethingHandler = services.get<IUpdateSomethingHandler>(SampleServices.Commands.IUpdateSomethingHandler);
                    await updateSomethingHandler.handleAsync(new UpdateSomethingCommand({
                        someId: firstSomething.someId,
                        someNewData: firstSomething.someData + ' | ' + new Date().toISOString()
                    }));

                    // const removeSomethingHandler = services.get<IRemoveSomethingHandler>(SampleServices.Commands.IRemoveSomethingHandler);
                    // await removeSomethingHandler.handleAsync(new RemoveSomethingCommand({
                    //     someId: firstSomething.someId
                    // }));
                } else {
                    const addSomethingHandler = services.get<IAddSomethingHandler>(SampleServices.Commands.IAddSomethingHandler);
                    await addSomethingHandler.handleAsync(new AddSomethingCommand({
                        someNewData: new Date().toISOString()
                    }));
                }

                records = await queries.getSomethingsAsync();
                res.send(JSON.stringify(records));
                
            } catch(e) {
                console.log("Encountered error: ", e)
                res.send(JSON.stringify(e));
            }
        });

        return app;

    }

}

File: sample-api/dotnet/src/startup.cs
using System;
using System.Linq;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json;
using Sample.API.DependencyInjection;
using Sample.Infrastructure.Persistence.ORM;
using Sample.Services.Commands;
using Sample.Services.Queries;

namespace Sample.API
{
    public class Startup
    {

        private readonly IConfiguration _configuration;

        public Startup(IConfiguration configuration) => _configuration = configuration;

        public void ConfigureServices(IServiceCollection services)
        {
            var postgres = new PostgresConnection();
            _configuration.Bind("postgres", postgres);
            var ormType = OrmType.FromName(_configuration.GetValue<string>("orm"));

            services.AddSampleApi(options => 
            {
                options.DomainCommands = true;
                options.DomainQueries = true;
                options.ORM = ormType;
                options.Postgres = postgres;
            });
        }

        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.Map("/favicon.ico", delegate { }); // Ignore favicon request(s)

            app.Run(async (context) =>
            {
                try 
                {

                    var queries = app.ApplicationServices.GetRequiredService<IQueries>();
                    var records = await queries.GetSomethingsAsync();
                    if (records.Any())
                    {
                        var firstSomething = records.First();

                        var updateSomethingHandler = app.ApplicationServices.GetRequiredService<IUpdateSomethingHandler>();
                        await updateSomethingHandler.HandleAsync(new UpdateSomethingCommand
                        {
                            SomeId = firstSomething.SomeId,
                            SomeNewData = firstSomething.SomeData + " | " + DateTime.UtcNow.ToShortTimeString()
                        });
                        
                        // var removeSomethingHandler = app.ApplicationServices.GetRequiredService<IRemoveSomethingHandler>();
                        // await removeSomethingHandler.HandleAsync(new RemoveSomethingCommand
                        // {
                        //     SomeId = firstSomething.SomeId
                        // });
                    } 
                    else
                    {
                        var addSomethingHandler = app.ApplicationServices.GetRequiredService<IAddSomethingHandler>();
                        await addSomethingHandler.HandleAsync(new AddSomethingCommand
                        {
                            SomeNewData = DateTime.UtcNow.ToShortTimeString()
                        });
                    }
                    
                    records = await queries.GetSomethingsAsync();
                    await context.Response.WriteAsync(JsonConvert.SerializeObject(records));
                
                }
                catch (Exception e)
                {
                    Console.WriteLine("Encountered error: {0}\n\n{1}", 
                        JsonConvert.SerializeObject(e.Message), 
                        JsonConvert.SerializeObject(e.StackTrace)
                    );
                    await context.Response.WriteAsync(string.Format("{0}\n\n{1}", 
                        JsonConvert.SerializeObject(e.Message), 
                        JsonConvert.SerializeObject(e.StackTrace)
                    ));
                }
            });
        }
    }
}
